articles/secret-key-cryptography-in-js/index.md

```diff
@@ -0,0 +1,138 @@
+```
+From the above, we define the module crypto which is a core module meaning. From there, we specify the algorithm we are going to use since crypto can support more than one algorithm. We shall use `AES`. We then come up with the neccessary parameters to get the shared key. A one time code is also needed in generating the cipher and the decipher. Based on the above, we have everything for encryption and decryption.
+
+#### David encrypting the message
+
+```javascript
+    function davidEncrypting(){
+    const davids_message = "Hello there";
+
+    cipher.on('readable',() => {
+        let _cipher_text = cipher.read()
+        if(_cipher_text){
+
+            cipher_text = _cipher_text.toString('hex');
+
+        }
+    });
+
+    cipher.write(davids_message);
+
+    cipher.end();
+   
+    };
+```
+
+When you are encrypting, you will need the message to be encrypted and from this instance, its coming from David. From there we need to listen to the readable event of the cipher since that is where we are going to get the cipher text. We write the message to the cipher and then close the cipher.
+
+#### Viktoria decrypting the message
+
+```javascript
+    decipher.on('readable',() => {
+
+        let _plain_text = decipher.read();
+
+        if(_plain_text){
+
+            console.log(_plain_text.toString('utf8'));
+            
+        };
+
+
+    });
+
+    decipher.write(cipher_text,'hex');
+
+    decipher.end();
+```
+
+Expected output
+
+```bash
+    Hello there
+```
+
+Decipher works as similar to the cipher. We still listen to the readable event and look if we have text which we convert to human readable format `UTF8`. If then we have the plain text, then we can show the output. We write the cipher text to the decipher using the `hex` format and the close the decipher. 
+
+### Drawback of secret key cryptography
+
+#### Key distirbution problem.
+Secret key cryptography used the same key for encrypting and decrypting the message. The key is usually transported in an insecure medium and therefore  questions the intergrity of the data in transit. An attacker with bad intentions can try hard to get the key and this shall mean that he or she can be able to encode and decode the message without meeting authorization requirements. 
+
+solutions to key distribution problem include:
+- Public key cryptography
+- Deffie Hellman key distribution algorithm
+
+### Existing platforms that use secret key cryptography
+
+Some of the platforms that use secret key cryptography in production include:
+
+- IBM
+- National Institute of Science and Technology (NIST)
+
+### Conclusion
+In this article we have looked at brief description of secret key cryptography, algorithms that employ secret key cryptography, libraries to use while implementing secret key cryptography in JavaScript, implementing secret key cryptography in JavaScript, drawback of secret key cryptography and platforms that employ secret key cryptography.
```

Comment:
"at a brief" not "at brief"
"the drawback" not "drawback"
"cryptography, and platforms that employs" <- oxford comma